account_asset_management/models/account_asset.py

```diff
@@ -646,35 +654,75 @@ def _compute_depreciation_line(
                     seq -= 1
             line_i_start = 0
 
-    def compute_depreciation_board(self):
+    def _unlink_asset_lines_whitout_move(self):
+        self.ensure_one()
+        line_obj = self.env["account.asset.line"]
+        domain = [
+            ("asset_id", "=", self.id),
+            ("type", "=", "depreciate"),
+            ("move_id", "=", False),
+            ("init_entry", "=", False),
+        ]
+        line_obj.search(domain).unlink()
 
+    def _get_asset_posted_lines(self):
```

Comment:
Why extract this into 2 methods for the same domain having more code lines at the end? And you are not adding any docstring for these new hook methods.

Reply:
I passed this on to two methods for reducing complexity.

account_asset_management/models/account_asset.py:657:5: C901 'AccountAsset.compute_depreciation_board' is too complex (19)